### ISSUE TITLE
some sort of type error 

### ISSUE THREAD

**Issue by Hussain-Shahid** [chatengine-io/react-chat-engine#27]
import {ChatEngine} from 'react-chat-engine'
  import './App.css';

function App() {
  return (
   <ChatEngine
      height="100vh"
      projectID="1a346ac3-aa55-4f49-aab1-2ccfe04b0598"
      userName="Hussainbutt"
      userSecret="1234" 
      
   />
  );
}

export default App;
![image](https://user-images.githubusercontent.com/48852087/113165998-cb92b200-925b-11eb-92c4-a24aff677758.png)

**Comment by alamorre** [comment]
I'll take a look and publish a fix within 1-2 hours.

**Comment by alamorre** [comment]
Please let me know if the issue goes away in **version 1.8.1** thanks a bunch! 😄

**Comment by Hussain-Shahid** [comment]
if you dont mind can you tell me why this issue occur i tried to resolve
this by myself but failed .

**Comment by alamorre** [comment]
Basically there was a race condition with loading the connection object, so in 1.8.1 it waits until the connection object has fully loaded. Again, please let me know if there are other issues - happy to fix them 👍👍

**Comment by Hussain-Shahid** [comment]
same issue

**Comment by alamorre** [comment]
<img width="1440" alt="Screen Shot 2021-03-31 at 10 35 52 AM" src="https://user-images.githubusercontent.com/15023107/113186738-07ab2000-9226-11eb-918a-d3497c9e2813.png">

```
import React, { Component } from 'react';
import {ChatEngine} from 'react-chat-engine'

export default class DirectChatsPage extends Component {
    render() {
        return (
            <ChatEngine
                height="100vh"
                projectID="1a346ac3-aa55-4f49-aab1-2ccfe04b0598"
                userName="Hussainbutt"
                userSecret="1234"
            />
        );
    }

}
```

This looks fine to me

**Comment by Hussain-Shahid** [comment]
use props username and enter the correct username then you will get the
error

**Comment by codeWithTyson** [comment]
i got the same error 
![Screenshot (31)](https://user-images.githubusercontent.com/68985954/113190701-7c279400-9211-11eb-956c-753f211e61a7.png)

![Screenshot (30)](https://user-images.githubusercontent.com/68985954/113190693-77fb7680-9211-11eb-9c31-cda9e8ba67f7.png)

**Comment by alamorre** [comment]
Lol I'm pushing out 1.8.2 which will handle Null and Undefined. Let me know if 1.8.2 works

1.8.X has a lot of powerful updates which will allow **TONS** of customizations. Please bare with me this will be worth it! 😁

**Comment by BinarySenseiii** [comment]
still not fixed i update t0 1.8.2 
![Screenshot (32)](https://user-images.githubusercontent.com/68985954/113191863-ec82e500-9212-11eb-8a25-5cfee37ba729.png)
![Screenshot (33)](https://user-images.githubusercontent.com/68985954/113191877-f1479900-9212-11eb-9396-fee075d8282c.png)


![Screenshot (34)](https://user-images.githubusercontent.com/68985954/113191841-e4c34080-9212-11eb-90af-b9c44a01ebe3.png)

**Comment by IZACK96** [comment]
I have tried version 1.8.2 and the userName error is gone, but now as i try to customize the chat feed with "renderChatFeed", i get this error
![Capture1](https://user-images.githubusercontent.com/21972713/113216667-f5a89c00-927c-11eb-9e1d-b9a0c793af64.PNG)

**Comment by alamorre** [comment]
@ anybody who wants to try 1.8.3 please let me know if the issues are going away. I ran into some on my own and I think i got it... 

`yarn add react-chat-engine@1.8.3`

**Comment by codeWithTyson** [comment]
> I have tried version 1.8.2 and the userName error is gone, but now as i try to customize the chat feed with "renderChatFeed", i get this error
> ![Capture1](https://user-images.githubusercontent.com/21972713/113216667-f5a89c00-927c-11eb-9e1d-b9a0c793af64.PNG)

hey maybe u did something wrong bcz in version 1.8.3 i didn't find any issue, n i recently customize chat feed too.

**Comment by basileLeroy** [comment]
![image](https://user-images.githubusercontent.com/72804463/113316037-3997b180-930e-11eb-90f2-14f4a44bdbdd.png)

It seems that this error message shown above only happens every once in a while for me.
 
###  **EDIT: The above issue seems to have been fixed with the 1.8.3 version**
___


![image](https://user-images.githubusercontent.com/72804463/113316509-b62a9000-930e-11eb-979c-760dfc70da73.png)

On the other hand, I seem to have problems when entering new messages. on enter/submit (same handler), I get the TypeError above.

![image](https://user-images.githubusercontent.com/72804463/113317029-410b8a80-930f-11eb-9a06-bfe3865d7e8c.png)
My console log seems to be showing it fine though..

___

My engine:
```
<ChatEngine
        height="100vh"
        userName="Tetris"
        userSecret="XXXXXX"
        projectID="3f78a1ff-b807-4a82-b8e1-aeab29b74a34"
        renderChatFeed={(chatroomProps) => <ChatFeed {... chatroomProps} /> }					
/>
```

My send message handler inside component (MessageForm):

```
const [value, setValue] = useState('');
const { creds, chatID } = props;
    
const handleSubmit = (event) => {
        event.preventDefault();
        const text = value.trim();
    
        if (text.length > 0) {
            sendMessage(creds, chatID, { text });
       }
       setValue('');
};
```

and my component:
```
<MessageForm {...props} chatID={activeChat} />
```


I hope this was enough info to find an answer! :) 
If not let me know, i am just a rookie!

**Comment by BinarySenseiii** [comment]
> ![image](https://user-images.githubusercontent.com/72804463/113316037-3997b180-930e-11eb-90f2-14f4a44bdbdd.png)
> 
> It seems that this error message shown above only happens every once in a while for me.
> 
> ### **EDIT: The above issue seems to have been fixed with the 1.8.3 version**
> ![image](https://user-images.githubusercontent.com/72804463/113316509-b62a9000-930e-11eb-979c-760dfc70da73.png)
> 
> On the other hand, I seem to have problems when entering new messages. on enter/submit (same handler), I get the TypeError above.
> 
> ![image](https://user-images.githubusercontent.com/72804463/113317029-410b8a80-930f-11eb-9a06-bfe3865d7e8c.png)
> My console log seems to be showing it fine though..
> 
> My engine:
> 
> ```
> <ChatEngine
>         height="100vh"
>         userName="Tetris"
>         userSecret="XXXXXX"
>         projectID="3f78a1ff-b807-4a82-b8e1-aeab29b74a34"
>         renderChatFeed={(chatroomProps) => <ChatFeed {... chatroomProps} /> }					
> />
> ```
> 
> My send message handler inside component (MessageForm):
> 
> ```
> const [value, setValue] = useState('');
> const { creds, chatID } = props;
>     
> const handleSubmit = (event) => {
>         event.preventDefault();
>         const text = value.trim();
>     
>         if (text.length > 0) {
>             sendMessage(creds, chatID, { text });
>        }
>        setValue('');
> };
> ```
> 
> and my component:
> 
> ```
> <MessageForm {...props} chatID={activeChat} />
> ```
> 
> I hope this was enough info to find an answer! :)
> If not let me know, i am just a rookie!

m facing the same issue

**Comment by alamorre** [comment]
Closed due to inactivity
